### PR TITLE
Add error messages to display in Frontend

### PR DESCRIPTION
**Pull request by hiongkaihan**
1. Added confirm password field to both frontend and backend when signing up
2. Added error message alert in frontend

### CODE REVIEW

File: frontend/src/components/LoginPage.js [hunk 1]
```diff
@@ -25,11 +29,15 @@ const LoginPage = (event) => {
         event.preventDefault();
         const formData = new FormData(event.target);
         const formDataObj = Object.fromEntries(formData.entries());
-        await userService.login(formDataObj);
-        navigate('/home')
+        const res = await userService.login(formDataObj)
+            .catch((err) => {
+                setAlertMessage(err.response.data.message);
```
Comment (ongweisheng) [inline frontend/src/components/LoginPage.js:34]: need to setAlertmessage(null) after a setTimeout function

File: frontend/src/components/SignupPage.js [hunk 2]
```diff
@@ -24,11 +27,15 @@ const SignupPage = (event) => {
...
-        navigate('/login')
+        const res = await userService.signup(formDataObj)
+            .catch((err) => {
+                setAlertMessage(err.response.data.message);
+            })
+        if (res) navigate('/login');
     }
     return (
         <>
+            {alertMessage ? (<AlertMessage onClose={() => setAlertMessage(null)} message={(alertMessage)}/>) : <div></div>}
```
Comment (ongweisheng) [inline frontend/src/components/SignupPage.js:38]: Same comments as in LoginPage.js

**Review by ongweisheng — CHANGES_REQUESTED** [review]
Some minor changes to improve on

File: frontend/src/components/LoginPage.js [hunk 3]
```diff
@@ -25,11 +29,15 @@ const LoginPage = (event) => {
...
+            })
+        if (res) navigate('/home');
     }
     return (
         <>
+            {alertMessage ? (<AlertMessage onClose={() => setAlertMessage(null)} message={(alertMessage)}/>) : <div></div>}
```
[the remark this answers is not in the document]
Comment (hiongkaihan) [reply frontend/src/components/LoginPage.js]: Done

File: frontend/src/components/LoginPage.js (hunk 1 above)
Comment (hiongkaihan) [reply frontend/src/components/LoginPage.js:34]: Done

File: frontend/src/components/SignupPage.js (hunk 2 above)
Comment (hiongkaihan) [reply frontend/src/components/SignupPage.js]: Done

File: frontend/src/components/AlertMessage.js [hunk 4]
```diff
@@ -1,12 +1,19 @@
 import Alert from 'react-bootstrap/Alert';
 
 const AlertMessage = ({onClose, message}) => {
+
+    if (!message) {
```
Comment (ongweisheng) [inline frontend/src/components/AlertMessage.js:5]: Maybe just check message === null? Don’t know if !null will crash stuff.

Comment (hiongkaihan) [reply frontend/src/components/AlertMessage.js]: Fixed

**Review by ongweisheng — APPROVED** [review]
LGTM

**Pull request merged**